staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/kubeopenapi.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schema
+
+import (
+	"k8s.io/kube-openapi/pkg/validation/spec"
+)
+
+// ToKubeOpenAPI converts a structural schema to go-openapi schema. It is faithful and roundtrippable.
+func (s *Structural) ToKubeOpenAPI() *spec.Schema {
```

Comment:
Adding this in parallel to `ToGoOpenAPI` is unfortunate. How will we avoid drift / bugs between the two? Is there a plan to eliminate one?

Reply:
yes, we will cut the other (less-critical) bit of go-openapi dependency in the publishing pipeline.